BookShelf/Startup.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.HttpsPolicy;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.EntityFrameworkCore;
+using BookShelf.Models;
+using BookShelf.BookService;
+
+namespace BookShelf
+{
+    public class Startup
+    {
+        public Startup(IConfiguration configuration)
+        {
+            Configuration = configuration;
+        }
+
+        public IConfiguration Configuration { get; }
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        public void ConfigureServices(IServiceCollection services)
+        {
+            //services.AddDbContext<BookContext>(opt =>
+            //    opt.UseInMemoryDatabase("BookList"));
+            services.AddSingleton<IBook, BookList>();
```

Comment:
It is strange to have IBook interface that is implemented  BookList. Commonly If you have IBook - Book will implement it. 

Singleton is a bad idea to use in the future. Only in case of real need. In all other cases use transient or scoped

Reply:
Anton, transient or scoped does not store the state of object between requests. Therefore, we will be unable to check the result of delete or update. I.e. we will not any changes in our list  of books, because it will be created from scratch.